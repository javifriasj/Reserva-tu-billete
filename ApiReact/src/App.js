import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const DEFAULT_QUERY = 'react';
const PATH_BASE = 'https://hn.algolia.com/api/v1';
const PATH_SEARCH = '/search';
const PARAM_SEARCH = 'query=';

var url = `${PATH_BASE}${PATH_SEARCH}?${PARAM_SEARCH}${DEFAULT_QUERY}`;

//const isSearched = (searchTerm) => (item) => !searchTerm || item.title.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase());

class App extends Component {
    constructor(props)
    {
      super(props);
      this.state =
      {
        result: null,
        searchTerm: DEFAULT_QUERY
      };

      this.setSearchTopStories = this.setSearchTopStories.bind(this);
      this.fetchSearchTopStories = this.fetchSearchTopStories.bind(this);
      this.onDimiss = this.onDimiss.bind(this);
      this.onSearchChange = this.onSearchChange.bind(this);
      this.onSearchSubmit = this.onSearchSubmit.bind(this);
    }

    onSearchSubmit(event) {
      const { searchTerm } = this.state;
      this.fetchSearchTopStories(searchTerm);
      event.preventDefault();
    }

    componentDidMount() {
      const {searchTerm} = this.state;
      this.fetchSearchTopStories(searchTerm);
    }

    fetchSearchTopStories(searchTerm) {
      fetch(`${PATH_BASE}${PATH_SEARCH}?${PARAM_SEARCH}${searchTerm}`)
      .then(response => response.json())
      .then(result => this.setSearchTopStories(result));
    }

    setSearchTopStories(result) {
      this.setState({result});
    }

    onSearchChange(event)
    {
      this.setState({searchTerm:event.target.value});
    }

    onDimiss(id)
    {
      const isNotId = item => item.objectID !== id;
      const updateList = this.state.result.hits.filter(isNotId);
      this.setState({
        result: { ...this.state.result, hits:updateList }
        });
    }

    render()
    {
      const {searchTerm, result} = this.state;
      return (
      <div className="APP">
        <Search value={searchTerm} onSubmit={this.onSearchSubmit} onChange={this.onSearchChange}>Search </Search>
        { result &&
        <Table list={result.hits} onDimiss={this.onDimiss}/>
        }
      </div>
    );
  }
}

const Search = ({value, onChange, onSubmit, children}) => {
    return (<form onSubmit={onSubmit}>{children}<input type="text" value={value} onChange={onChange}/> <button type="submit">{children}</button></form>);
}

const Table = ({list, onDimiss}) => {
  return(
      <div>
      { list.map(item =>
          <div key={item.objectID} className="table-row">
            <span>{item.title}</span>
            <span><Button onClick={()=>onDimiss(item.objectID)} type="button">Dimiss</Button></span>
          </div>
      )}
    </div>
  );
}

const Button = ({onClick, className='', children}) => {
    return (<button onClick={onClick} className={className} type="button">{children}</button>);
}

export default App;
export {Button, Search, Table};
