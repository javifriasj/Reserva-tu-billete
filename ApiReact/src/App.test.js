import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import { shallow } from 'enzyme';
import app, {Search, Button, Table} from './App';

describe('Search', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(<Search>Search</Search>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Button', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Give me more</Button>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(<Button>Give me more</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('table', () => {

  const props = {
    list:[
      {title:'1', author:'1', num_comments:1, points:2, objectID:'y'},
      {title:'2', author:'2', num_comments:1, points:2, objectID:'z'}
    ]
  }

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table { ...props }/>,div);
  });

  test('snapshots', () => {
    const component = renderer.create(<Table { ...props }/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Shows two items in list', () => {
    const element = shallow(<Table { ...props }/>);
    expect(element.find('.table-row').length).toBe(2);
  });

});
