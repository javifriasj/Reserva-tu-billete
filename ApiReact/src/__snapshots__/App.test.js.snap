exports[`Button snapshots 1`] = `
<button
  className=""
  onClick={undefined}
  type="button">
  Give me more
</button>
`;

exports[`Search snapshots 1`] = `
<form
  onSubmit={undefined}>
  Search
  <input
    onChange={undefined}
    type="text"
    value={undefined} />
   
  <button
    type="submit">
    Search
  </button>
</form>
`;

exports[`table snapshots 1`] = `
<div>
  <div
    className="table-row">
    <span>
      1
    </span>
    <span>
      <button
        className=""
        onClick={[Function]}
        type="button">
        Dimiss
      </button>
    </span>
  </div>
  <div
    className="table-row">
    <span>
      2
    </span>
    <span>
      <button
        className=""
        onClick={[Function]}
        type="button">
        Dimiss
      </button>
    </span>
  </div>
</div>
`;
